billing-service/src/main/java/org/egov/demand/service/BillServicev2.java

```diff
@@ -284,26 +288,48 @@ public BillResponseV2 generateBill(GenerateBillCriteria billCriteria, RequestInf
 				.tenantId(billCriteria.getTenantId())
 				.email(billCriteria.getEmail())
 				.consumerCode(consumerCodes)
-				.isPaymentCompleted(false)
 				.receiptRequired(false)
 				.demandId(demandIds)
 				.build();
 
 		/* Fetching demands for the given bill search criteria */
-		List<Demand> demands = demandService.getDemands(demandCriteria, requestInfo);
+		List<Demand> demandsWithMultipleActive = demandService.getDemands(demandCriteria, requestInfo);
+
+		if (demandsWithMultipleActive.isEmpty()) {
+			throw new CustomException(EG_BS_BILL_NO_DEMANDS_FOUND_KEY, EG_BS_BILL_NO_DEMANDS_FOUND_MSG);
+		}
+
+		//filter the demands which are fully paid
+		demandsWithMultipleActive = demandsWithMultipleActive.stream().filter(demand -> !demand.getIsPaymentCompleted()).collect(Collectors.toList());
+
+		List<Demand> demands = filterMultipleActiveDemands(demandsWithMultipleActive);
 
 		List<BillV2> bills;
 
 		if (!demands.isEmpty())
 			bills = prepareBill(demands, requestInfo);
 		else
-			throw new CustomException(EG_BS_BILL_NO_DEMANDS_FOUND_KEY, EG_BS_BILL_NO_DEMANDS_FOUND_MSG);
+			return getBillResponse(Collections.emptyList());
```

Comment:
this single line of change is enough

Reply:
@kavi-egov 
As you made the changes this PR can be closed ?Please update.

Reply:
I already made changes in develop to cover all. pleasee close the PR